lib/teleterm/daemon/daemon.go

```diff
@@ -212,13 +262,63 @@ func (s *Service) createGateway(ctx context.Context, params CreateGatewayParams)
 	return gateway, nil
 }
 
-func (s *Service) onExpiredGatewayCert(ctx context.Context, gateway *gateway.Gateway) error {
-	cluster, err := s.ResolveCluster(gateway.TargetURI())
+// reissueGatewayCerts tries to reissue gateway certs.
+func (s *Service) reissueGatewayCerts(ctx context.Context, g *gateway.Gateway) error {
+	cluster, err := s.ResolveCluster(g.TargetURI())
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	clusterURI, err := uri.ParseClusterURI(g.TargetURI())
 	if err != nil {
 		return trace.Wrap(err)
 	}
+	rootClusterURI := clusterURI.GetRootClusterURI().String()
```

Comment:
I am wondering if it would be better to `s.ResolveCluster` again after relogin (moving inside `reissueGatewayCert`)?